### PR TITLE
feat: add support for field metdata

### PR DESCRIPTION
**Pull request by westonpace**
Previously we had just extracted the extension name from the field and discarded the rest of the metadata.  Unfortunately, these fields are used to parameterize extension types.  For example:

```
// A fixed shape tensor extension type for tensors with shape (2, 3)
metadata: {
    "ARROW:extension:name": "arrow.fixed_shape_tensor",
    "ARROW:extension:metadata": "{\"shape\":[2,3]}",
},
```

This PR keeps all of the metadata information.  This also allows for a lossless transformation when users have their own custom field metadata.

### CODE REVIEW

File: rust/src/datatypes/field.rs [hunk 1]
```diff
@@ -546,30 +548,33 @@ impl TryFrom<ArrowField> for Field {
...
+                (key.clone(), string_value)
+            })
+            .collect();
+        if !field.extension_name.is_empty() {
+            lance_metadata.insert(
+                "ARROW:extension:name".to_string(),
+                field.extension_name.clone(),
+            );
+        }
+        // TODO (check extension_name)
```
Comment (wjones127) [inline rust/src/datatypes/field.rs:571]: What's this TODO? can it be removed?

**Review by wjones127 — COMMENTED** [review]
These changes look good. Do you want to add a test in Python for the repro I wrote in #1170?

File: rust/src/datatypes/field.rs (hunk 1 above)
Comment (westonpace) [reply rust/src/datatypes/field.rs]: Ah, sorry, forgot to remove this (it's resolved by the if check above it).  I have now removed this comment.

**Comment by westonpace** [comment]
> Do you want to add a test in Python for the repro I wrote in https://github.com/lancedb/lance/issues/1170?

I added fixed shape tensors to the `test_roundtrip_types` in `test_lance.py`

**Review by wjones127 — APPROVED** [review]

**Pull request merged**